mini.go

```diff
@@ -157,15 +176,43 @@ func Poster(w http.ResponseWriter, r *http.Request) {
 	id := Id(point.AdvertiserID)
 	ip := r.RemoteAddr
 
-	//store the data from the struct to the sql databases
-	_, errs = db.Exec("INSERT INTO clicks(id, advertiser_id, site_id, ip, ios_ifa, google_aid, windows_aid ) VALUES(?, ?, ?, ?, ?, ?, ?)",
+	//store the data from the struct to the sql databases and log the error or latency time
+	QueryStart := time.Now()
+
+	_, errs = db.Exec("INSERT INTO clicks(id, advertiser_id, site_id, ip, ios_ifa, google_aid, windows_aid) VALUES(?, ?, ?, ?, ?, ?, ?)",
 		id, point.AdvertiserID, point.SiteID, ip, point.IosIfa, point.GoogleAid, point.WindowsAid)
 
 	if errs != nil {
-		fmt.Println(errs)
+		errString := "sorry, there is an error"
+		response(w, errString, "", http.StatusInternalServerError)
+		errString = fmt.Sprintf("database connection error : %s", errs)
+		logWriter.Err(errString)
 		return
+	}
+
+	responseTime("the time for inserting data to clicks table is ", QueryStart)
+
+	//sucess and log the request latency
+	response(w, "", id, http.StatusOK)
+	responseTime("the time for this Post request is ", RequestStart)
+}
+
+//report the query / request latency
+func responseTime(message string, startTime time.Time) {
+	responseDuration := time.Since(startTime)
+	logWriter.Info(message + responseDuration.String())
+}
+
+//write the post reponse (faliure /success) to the client in  Json format
+func response(w http.ResponseWriter, errMessage string, id string, status int) {
+	w.WriteHeader(status)
+
+	validate := PostResponses{errMessage, id, strconv.Itoa(status)}
+	bytes, errs := json.Marshal(&validate)
+	if errs != nil {
+		fmt.Println(errs) // this errors is only for execution no need to output to user
 	} else {
-		io.WriteString(w, id)
+		io.WriteString(w, string(bytes))
```

Comment:
does not really matter but you could also do this a little more simply as:
`w.Write(bytes)`